### ISSUE TITLE
LHS hit navigator search doesn't work (maybe?)

### ISSUE THREAD

**Issue by phraenquex** [m2ms/fragalysis-frontend#1371]
This snapshot even recreates the search string (CX1134B) - but the LHS list doesn't shrink down to just that hit (the behaviour I expected/remembered).

https://fragalysis.xchem.diamond.ac.uk/viewer/react/projects/26/26

**Comment by boriskovar-m2ms** [comment]
This unfortunately works as designed. So many hits are visible in the hit navigator because they're visible in the NGL View. The rules for visibility and ordering of the LHS are so complicated that even I need to painstakingly debug the code to find out it's behaving in line with spec. For normal end user the behavior seems buggy or random at best. There is even a ticket somewhere, where the need to for some kind of simplification or maybe some feedback why hit is displayed to the user is described but I'm not able to find it.

**Comment by phraenquex** [comment]
First thing:  give visibility to user to why something is on there.  e.g. give a eye-type icon which will reveal that.

**Comment by mwinokan** [comment]
@boriskovar-m2ms, @phraenquex please take a look at my new mockup in the [root comment of this ticket](https://github.com/m2ms/fragalysis-frontend/issues/1371#issue-2170313990)

**Comment by phraenquex** [comment]
@mwinokan yes nice!  Plough ahead and get @matej-vavrek to do it:  it will obviously end up evolving, but we won't know how until it's in place.

(You'll need a "Search now" button.)

**Comment by phraenquex** [comment]
@mwinokan still missing:  where does the search "go".  My suggestion:  it creates a tag, called `<timestamp> <user>`  

- Needs a way to call up the search dialog that generated the tag.  My suggestion:  make the existing search icon red (?) and bring up the original search icon when the tag is active;  and if you then hit "Search", create a new tag
- You'll have to save the search config somewhere - I think tags have a free-form field, that can hold relevant json

(Of course, the question of _how_ to find your tag is then scooted off to the (still missing?) ticket for reworking the Tag navigator.  If you create that ticket, I can braindump there - but I think it should just be a table modal.)

**Comment by mwinokan** [comment]
I take your point @phraenquex, in that if you've done a complicated search you might want to repeat it eventually. 

However, I don't think the tags are the right home for this. It would make much more sense to me to have a "search history" table in that new modal, that allows you to repeat searches. Before then, the user can also use existing functionality (tags / snapshots) to save a certain state so I'm not sure it's crucial. New ticket: #1713

**Comment by mwinokan** [comment]
Some discussion today,

@phraenquex is wondering how much of each search query and search result needs to be stored in the backend, and how.

@boriskovar-m2ms says that tags are appropriate to store the result of a search

@mwinokan suspects that an extra model will be needed to store the search query, so that it can be repeated if there is new data available @boriskovar-m2ms says that if the b/e does not need to know about the query then it can be just in JSON of the tags

@phraenquex wonders if some hierarchy / relations between previous searches is needed

First steps will be to implement the advanced search queries, and then afterwards think about the backend integration

**Comment by kaliif** [comment]
If the backend needs to interact with the search, it almost certainly needs to be a new database object.

**Comment by mwinokan** [comment]
Work now included as part of the LHS & RHS unification: https://github.com/m2ms/fragalysis-frontend/issues/1417
